lisa/operating_system.py

```diff
@@ -554,33 +574,21 @@ def _get_os_version(self) -> OsVersion:
 
 
 class Redhat(Fedora):
-    # pattern for expired RHUI client certificate issue
-    __rhui_error_pattern = re.compile(
-        r"([\w\W]*?)SSL peer rejected your certificate as expired.*", re.MULTILINE
-    )
-
     @classmethod
     def name_pattern(cls) -> Pattern[str]:
         return re.compile("^rhel|Red|AlmaLinux|Scientific|acronis|Actifio$")
 
     def _initialize_package_installation(self) -> None:
-        # older images cost much longer time when update packages
-        # smaller sizes cost much longer time when update packages, e.g.
-        #  Basic_A1, Standard_A5, Standard_A1_v2, Standard_D1
-        # redhat rhel 7-lvm 7.7.2019102813 Basic_A1 cost 2371.568 seconds
-        # redhat rhel 8.1 8.1.2020020415 Basic_A0 cost 2409.116 seconds
-        cmd_result = self._node.execute("yum -y update", sudo=True, timeout=3600)
+        cmd_result = self._node.execute("yum makecache", sudo=True)
+        os_version = self._get_os_version()
         # we will hit expired RHUI client certificate issue on old RHEL VM image
         # use below solution to resolve it
         # refer https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/redhat/redhat-rhui#azure-rhui-infrastructure # noqa: E501
-        if cmd_result.exit_code != 0 and self.__rhui_error_pattern.match(
-            cmd_result.stdout
-        ):
+        if "Red Hat" == os_version.vendor and cmd_result.exit_code != 0:
```

Comment:
It doesn't need to check if it's Red Hat, because it's in RedHat class.

Reply:
but centos also uses the same code?

Reply:
Yes, It's intent to share the code. If CentOS or other derives has different behaviors, we can handle them in CentOS or other subclasses.

Reply:
then it is still necessary to have a check whether it is rhel or not.